Scripts/symbolic_hermitians/HermitianUtils.py

```diff
@@ -70,16 +93,19 @@ def anticommutator(self, HermitianA, HermitianB):
         return self
 
     def conjugate(self):
-        self.H = Conjugate(self.H)
+        for i in range(self.size):
+            for j in range(i, self.size):
+                self.H[i,j] = conjugate(self.H[i,j])
         return self
     
     # return the length of the SU(n) vector
-    def SU_vector_magnitude(self):
+    def SU_vector_magnitude2(self):
         # first get the sum of the square of the off-diagonal elements
         mag2 = 0
         for i in range(self.size):
             for j in range(i+1,self.size):
-                mag2 += self.H[i,j]*self.H[j,i]
```

Comment:
I don't see how it would - looking at the resulting expressions, they were analytically real. But the real and imaginary parts is clear enough, too, so I'll just leave it, I think.

Reply:
I seem to recall the resulting expression for `H[i,j] * H[j,i]` had a nonzero imaginary component. I would bet money that's because in the code generation script, when doing `HSI  = (N-Nbar.conjugate())`, `H[i,j]` was conjugated and `H[j.i]` wasn't ...

Reply:
conjugate() turned out not to have been the issue responsible for this ...